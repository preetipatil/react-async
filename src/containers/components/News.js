import React from 'react';
import NewsItem from './NewsItem';
import PropTypes from 'prop-types';

export function NoNews() {
  return <h3>No News Available...</h3>;
}

let News = ({ news }) => {
  if (news) {
    return news.map((article, index) => (
      <div key={`${index}`}>
        <NewsItem article={news[index]} />
      </div>
    ));
  } else {
    return <NoNews />;
  }
};

News.propTypes = {
  news: PropTypes.array,
};
export default News;
