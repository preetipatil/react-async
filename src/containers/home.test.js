import React from 'react';
import { shallow } from 'enzyme';
import Home, { Loading, Error } from './home';
import News from './components/News';

describe('Home Page Test', () => {
  it('should renders News correctly', () => {
    const props = {
      newsFeed: { loading: false, newsfeed: [], error: null },
      loadNews: () => {},
    };
    const tree = shallow(<Home.WrappedComponent {...props} />);
    expect(tree).toMatchSnapshot();
    expect(tree.containsMatchingElement(<News />)).toEqual(true);
  });

  it('should renders Loading when APi is loading', () => {
    const props = {
      newsFeed: { loading: true, newsfeed: [], error: null },
      loadNews: () => {},
    };
    const tree = shallow(<Home.WrappedComponent {...props} />);
    //expect(tree).toMatchSnapshot();
    expect(tree.containsMatchingElement(<Loading />)).toEqual(true);
  });

  it('should renders Error when APi is failed to load data', () => {
    const props = {
      newsFeed: { loading: false, newsfeed: [], error: 'Failed to load' },
      loadNews: () => {},
    };
    const tree = shallow(<Home.WrappedComponent {...props} />);
    //expect(tree).toMatchSnapshot();
    expect(tree.containsMatchingElement(<Error />)).toEqual(true);
  });
});
