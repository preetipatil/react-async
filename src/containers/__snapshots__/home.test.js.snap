// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Test should renders News correctly 1`] = `
<News
  news={Array []}
/>
`;
